internal/app/machined/pkg/runtime/v1alpha1/platform/vmware/vmware_amd64.go

```diff
@@ -119,51 +119,51 @@ func (v *VMware) Configuration(context.Context) ([]byte, error) {
 
 		extraConfig := rpcvmx.NewConfig()
 
-		// try to fetch `talos.config` from OVF
-		log.Printf("trying to find '%s' in OVF env", constants.VMwareGuestInfoConfigKey)
+		// try to fetch `talos.config` from plain extraConfig (ie, the old behavior)
+		log.Printf("trying to find '%s' in extraConfig", constants.VMwareGuestInfoConfigKey)
 
-		config, err := readConfigFromOvf(extraConfig, constants.VMwareGuestInfoConfigKey)
+		config, err := readConfigFromExtraConfig(extraConfig, constants.VMwareGuestInfoConfigKey)
 		if err != nil {
 			return nil, err
 		}
 
-		if config != nil {
+		if (config != nil) && (len(config) > 0) {
 			return config, nil
 		}
 
-		// try to fetch `userdata` from OVF
-		log.Printf("trying to find '%s' in OVF env", constants.VMwareGuestInfoFallbackKey)
+		// try to fetch `userdata` from plain extraConfig (ie, the old behavior)
+		log.Printf("trying to find '%s' in extraConfig", constants.VMwareGuestInfoFallbackKey)
 
-		config, err = readConfigFromOvf(extraConfig, constants.VMwareGuestInfoFallbackKey)
+		config, err = readConfigFromExtraConfig(extraConfig, constants.VMwareGuestInfoFallbackKey)
 		if err != nil {
 			return nil, err
 		}
 
-		if config != nil {
+		if (config != nil) && (len(config) > 0) {
```

Comment:
I think it might be better to filter out empty `config` in the `readConfigFrom*` functions and return nil, this will also fix confusing log messages if the key is found, but empty

Reply:
I had trouble to choose these two approaches, you convinced me. Fixed